packages/wrangler/src/experimental-assets.ts

```diff
@@ -413,6 +420,12 @@ export function verifyMutuallyExclusiveAssetsArgsOrConfig(
 			"Cannot use Experimental Assets and tail consumers in the same Worker. Tail Workers are not yet supported for Workers with assets."
 		);
 	}
+
+	if (!(args.script || config.main) && config.experimental_assets?.binding) {
+		throw new UserError(
+			"Cannot use Experimental Assets with a binding in an assets-only Worker."
```

Comment:
We always want to follow up with an action that users can take to solve the error

```suggestion
			"Cannot use Experimental Assets with a binding in an assets-only Worker.\n" +
			`Please specify the path to the entry point for your Worker either in your configuration file ("main"), or as an argument to your wrangler command.`
```